asyncnsq/tcp/connection.py

```diff
@@ -114,7 +115,8 @@ def queue(self):
 
     def close(self):
         """Close connection."""
-        self._do_close()
+        if not self.closed:
```

Comment:
self._do_close will check if self.closed.  so this is unnecessary.

Reply:
I noticed some unclosed connections in TCP writer. Actually that was cause by the auto-reconnect issue.
It was a uncleaned check.

Reply:
so we need a more elegant way to do this.
and according your code, this piece of code seems equal with the original.
if you can check it.

Reply:
It is not really necessary. I removed it in the latest commit.
We just need to add a check condition in the auto-reconnect function